src/xwayland/xwm/mod.rs

```diff
@@ -2135,9 +2135,14 @@ fn handle_event<D: XwmHandler + 'static>(
                     }
                 }
                 x if x == xwm.atoms.WM_CHANGE_STATE => {
+                    let data = msg.data.as_data32();
                     if let Some(surface) = xwm.windows.iter().find(|x| x.window_id() == msg.window).cloned() {
                         drop(_guard);
-                        state.minimize_request(xwm_id, surface);
+                        match data[0] {
+                            1 => state.unminimize_request(xwm_id, surface),
```

Comment:
Out of curiosity: What is value 2?

Reply:
I was wondering that too, but strangely it isn't mentioned in https://x.org/releases/X11R7.6/doc/xorg-docs/specs/ICCCM/icccm.html:

```
WithdrawnState		0
NormalState		1
IconicState		3 
```

Maybe it was used for something historically.